src/sentry/incidents/endpoints/serializers.py

```diff
@@ -244,6 +247,38 @@ def validate_target_type(self, target_type):
                 % [item.value for item in AlertRuleTriggerAction.TargetType]
             )
 
+    def validate(self, attrs):
+        if ("target_type" in attrs) != ("target_identifier" in attrs):
+            raise serializers.ValidationError(
+                "targetType and targetIdentifier must be passed together"
+            )
+        target_type = attrs.get("target_type")
+        access = self.context["access"]
+        identifier = attrs.get("target_identifier")
+        if target_type == AlertRuleTriggerAction.TargetType.TEAM:
+            try:
+                team = Team.objects.get(id=identifier)
+            except Team.DoesNotExist:
+                raise serializers.ValidationError("Team does not exist")
+            if not access.has_team(team):
+                raise serializers.ValidationError("You do not have access to this team")
```

Comment:
Are we worried about leaking information that the team exists but you don't have access to it? Should the not found & access denied error messages be combined?

Reply:
I'm not really sure if we care... I think teams are public information? I don't mind just changing it though